src/config.js

```diff
@@ -149,14 +150,23 @@ export default class Config {
 		}
 
 		// Try to retrieve it from the source object.
-		if ( source && ( typeof source[ name ] != 'undefined' ) ) {
-			return source[ name ];
+		if ( source && ( typeof source[ configurationName ] != 'undefined' ) ) {
+			return source[ configurationName ];
 		}
 
-		// If not found, take it from the definition.
+		let valueFromDefinition;
+
+		// If not found, try to take it from the definition.
 		if ( this.definition ) {
-			return this.definition[ name ];
+			valueFromDefinition = this.definition.get( name );
 		}
+
+		// If still not found throw an error
+		if ( typeof valueFromDefinition == 'undefined' ) {
+			throw new CKEditorError( 'undefined-configuration-name', { name } );
```

Comment:
The name of the error should start with class name. E.g. `config-undefined-option`. Additionally, it must contain a human readable description of the error and, if more documentation is needed than that, an `@error` API doc. See some other examples across the codebase.